### PR TITLE
 Verify in unit test that BigQuery executing project is used.

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @dhalperi

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java [hunk 1]
```diff
@@ -238,27 +241,42 @@ public FakeJobService pollJobReturns(Object... pollJobReturns) {
...
         throws IOException, InterruptedException {
-      startJob();
+      startJob(jobRef);
     }
 
     @Override
     public Job pollJob(JobReference jobRef, int maxAttempts)
         throws InterruptedException {
+      if (!Strings.isNullOrEmpty(executingProject)) {
+        Preconditions.checkArgument(
```
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java]: please static import functions from Preconditions

Comment (peihe) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/BigQueryIOTest.java]: done

**Comment by peihe** [comment]
PTAL

**Comment by peihe** [comment]
updated the project name to "non-executing-project"

**Pull request closed without merging**